### PR TITLE
[Java] [Kotlin] ignore unsupported schemes instead of throwing exception

### PR DESCRIPTION
**Pull request by tiffmaelite**
Partially revert changes of #14876 and replace them with a new implementation which fixes the underlying issue while avoiding changing behavior by introducing potentially unexpected exception-throwing situations. As a result, the previous behavior of silently ignoring unsupported schemes is kept.

### PR checklist
 
- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [X] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [X] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Java: @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)

Kotlin: @jimschubert (2017/09) [❤️](https://www.patreon.com/jimschubert), @dr4ke616 (2018/08) @karismann (2019/03) @Zomzog (2019/04) @andrewemery (2019/10) @4brunu (2019/11) @yutaka0m (2020/03) @stefankoppier (2022/06)

### CODE REVIEW

File: samples/client/petstore/kotlin-retrofit2-kotlinx_serialization/src/main/kotlin/org/openapitools/client/infrastructure/ApiClient.kt [hunk 1]
```diff
@@ -72,11 +72,16 @@ class ApiClient(
         authNames: Array<String>
     ) : this(baseUrl, okHttpClientBuilder) {
         authNames.forEach { authName ->
-            val auth = when (authName) {
-                "petstore_auth" -> OAuth(OAuthFlow.implicit, "http://petstore.swagger.io/api/oauth/dialog", "", "write:pets, read:pets")"api_key" -> ApiKeyAuth("header", "api_key")
```
Comment (tiffmaelite) [inline samples/client/petstore/kotlin-retrofit2-kotlinx_serialization/src/main/kotlin/org/openapitools/client/infrastructure/ApiClient.kt:76]: I think that this was also incorrect kotlin code due to the lack of line break before "api_key"

Comment (wing328) [reply samples/client/petstore/kotlin-retrofit2-kotlinx_serialization/src/main/kotlin/org/openapitools/client/infrastructure/ApiClient.kt:76]: agreed. the CI should have caught the issue. let me take a look tomorrow or later this week

Comment (wing328) [reply samples/client/petstore/kotlin-retrofit2-kotlinx_serialization/src/main/kotlin/org/openapitools/client/infrastructure/ApiClient.kt:76]: the code still compiles in the latest master so the CI is doing the right thing

Comment (wing328) [reply samples/client/petstore/kotlin-retrofit2-kotlinx_serialization/src/main/kotlin/org/openapitools/client/infrastructure/ApiClient.kt:76]: i cannot come up with a way off the top of my head to catch this issue easily moving forward. will merge this for the time being and later add a test (hopefully)

File: modules/openapi-generator/src/main/resources/kotlin-server/libraries/ktor/_principal.mustache [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-{{#isApiKey}}
+{{/isBasicBasic}}{{^isBasicBasic}}{{#isApiKey}}
 val principal = call.authentication.principal<ApiPrincipal>()!!
-{{/isApiKey}}
-{{#isOAuth}}
+{{/isApiKey}}{{^isApiKey}}{{#isOAuth}}
 val principal = call.authentication.principal<OAuthAccessTokenResponse>()!!
-{{/isOAuth}}
+{{/isOAuth}}{{^isOAuth}}
+val principal = null!!
```
Comment (pschichtel) [inline modules/openapi-generator/src/main/resources/kotlin-server/libraries/ktor/_principal.mustache:9]: is this intentionally typed as Nothing ?

**Pull request merged**